core/modules/editor/operations/text/make-markdown-link.js

```diff
@@ -0,0 +1,37 @@
+/*\
+title: $:/core/modules/editor/operations/text/make-markdown-link.js
+type: application/javascript
+module-type: texteditoroperation
+
+Text editor operation to make a markdown link
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+exports["make-markdown-link"] = function(event,operation) {
+	if(operation.selection) {
+		if(event.paramObject.text.includes("://")) {
+			operation.replacement = "[" + operation.selection + "](" + event.paramObject.text + ")";
+		} else {
+			operation.replacement = "[" + operation.selection + "](#" + event.paramObject.text.replaceAll(" ", "%20") + ")";
```

Comment:
I'm not 100% sure, if this replacement is enough. What if there are Cyrillic characters in the URL. ... We do have a js function to convert text into URL strings, but I don't know the name atm

Reply:
I have tested it with such a link:

```md
[link](#Српски%20језик)
```

It works at least in Firefox 100.0 on Linux. When I hover over the resulting link, I see: <file:///home/max/Downloads/Wiki/markdown-test.html#Српски језик>. When I copy and paste the resulting link, I get:
<file:///home/max/Downloads/Wiki/markdown-test.html#%D0%A1%D1%80%D0%BF%D1%81%D0%BA%D0%B8%20%D1%98%D0%B5%D0%B7%D0%B8%D0%BA>.

I think the space character is special here because a markdown link can look like this:

```md
[example link](http://example.com/ "With a Title")
```
The (first?) space is interpreted as a delimiter between URL and the title attribute. That's why the the URL must not contain an unencoded space.

Reply:
ok